lms/security.py

```diff
@@ -217,6 +218,14 @@ def permits(self, request, _context, permission):
         return _permits(self.identity(request), permission)
 
 
+@dataclass
+class EmailPreferencesIdentity:
+    """The identity class used by EmailPreferencesSecurityPolicy."""
+
+    h_userid: str
+    tag: Optional[str] = None
```

Comment:
It's frustrating that `tag` has to be optional here because the code could be simpler if `tag` was always present, but Pyramid's `AuthTktCookieHelper` only stores a user ID in the cookie it doesn't provide a nice way to also store a tag, so when the authentication comes from the cookie we only have the user ID not the tag. As it happens we don't need the tag from the cookie anyway, so it's fine.